pkg/cloud/services/eks/cluster.go

```diff
@@ -114,6 +114,10 @@ func (s *Service) reconcileCluster(ctx context.Context) error {
 		return errors.Wrap(err, "failed reconciling cluster config")
 	}
 
+	if err := s.reconcileEKSEncryptionConfig(cluster.EncryptionConfig); err != nil {
```

Comment:
We will need some more logic around this call.  For example, if i am creating a new cluster then the encryption will have been enabled in `createCluster` and then it will update the encryption config again in `reconcileEKSEncryptionConfig`. 

What do you think to this:
1. We have the existing configuration  from AWS [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/dad36a88f506a7dd05ba19a5e5ff25bb69bdf9a0/pkg/cloud/services/eks/cluster.go#L51)
2.  The existing encryption config is passed into `reconcileEKSEncryptionConfig` (as you are doing now)
3. The desired encryption configuration from the AWSManagedControlPlane is coming from [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/dad36a88f506a7dd05ba19a5e5ff25bb69bdf9a0/pkg/cloud/services/eks/cluster.go#L483)
4. So we have the actual and desired config at this point
5.  We add some logic to `reconcileEKSEncryptionConfig` based on actual/desired that determines if there is a difference and what action needs to be taken (i.e. no action, AssociateEncryptionConfig or raise error)

What do you think?

Reply:
I think it sounds fair, i will take into consideration about updating the encryption config again (first in createCluster, then in reconcileEKSEncryptionConfig)

Reply:
in what scenario, we can have error in `reconcileEKSEncryptionConfig`

Reply:
An error shouldn't happen with the validating webhook. However, the Webhooks could be disabled and it's probably best to check for the incorrect update scenario to be safe.